docs/architecture.md

```diff
@@ -0,0 +1,121 @@
+
+`gcloud auth login`
+
+>**NOTE:** If this is your first time deploying, you should also run `gcloud init` and reinitialize your configuration. 
+
+
+## Secure Defaults
+
+When using the cli defaults, `gkekitctl create`, a Private Cluster in a Standalone VPC with the settings below is built. A config file can be used to customize the install, the spec and examples are kept in this [folder](cli/samples). Definitions of each configuration choice in the spec are stored in the [configuration.md](docs/configuration.md) doc. The rest of this doc goes over all the non-optional security features every cluster is bootstraped with.
+
+Private clusters allow you to isolate nodes from the public internet. Every GKE cluster has a Kubernetes API server that is managed by the control plane (master). In private clusters, the control plane's VPC network is connected to your cluster's VPC network with VPC Network Peering. Your VPC network contains the cluster nodes, and a separate Google Cloud VPC network contains your cluster's control plane. The control plane's VPC network is located in a project controlled by Google. Traffic between nodes and the control plane is routed entirely using internal IP addresses.
+
+Private clusters also restrict access to the internet by default. A NAT gateway of some form needs to be deployed should you want to enable outbound internet access from pods running in private clusters. Keep in mind this includes base container images not stored in the container registries that Google cloud maintains. In the following examples, a Google Cloud Nat Gateway is deployed alongs side the GKE clusters. 
+
+The GKE Cluster Install step in this repository will build a GKE Private cluster with access to the control plane with the following configuration:
+
+* [Public Endpoint Cluster]() - The cluster can be deployed with public access to the master endpoints therefore eliminating the need for a bastion host. Doing so configures the cluster as follows:
+  * Public endpoint for control plane is enabled
+  * Nodes receive private IP addresses
+  * Restricts access to addresses specified in the authorized networks list
+  * Authorized networks range can contain internal or public IP addresses
+
+Should you want to expose the control plane with a private IP, set `privateEndpoint: true` in your config file. When you choose that setting, the following is configured: 
+
+* [Private Endpoint Cluster]():
+  * Public endpoint for control plane is disabled
+  * Nodes receive private IP addresses
+  * Restricts access to addresses specified in the authorized networks list
+  * Authorized networks range must containe internal IP addresses
+
+The following best practices are also enforced as part of the cluster build process:
```

Comment:
Can we add a note that all of these are enabled and (currently) cannot be turned off?

Reply:
"The following security features of GKE are enabled at default and are not exposed to the user as optional:"